File: easy/1935_maximum_number_of_words_you_can_type/solution.cpp
#include <string>
#include <unordered_set>

class Solution {
public:
  int canBeTypedWords(std::string text, std::string brokenLetters) {
    std::unordered_set<char> brokenLettersSet(brokenLetters.begin(),
                                              brokenLetters.end());

    int wordsCount = 0;

    for (int i = 0; i < text.size(); i++) {
      if (brokenLettersSet.find(text[i]) != brokenLettersSet.end())
        while (i < text.size() - 1 && text[++i] != ' ') {
        }
      else if (text[i] == ' ' || i == text.size() - 1)
        ++wordsCount;
    }

    return wordsCount;
  }
};

File: easy/1935_maximum_number_of_words_you_can_type/solution.py
class Solution:
    def canBeTypedWords(self, text: str, broken_letters: str) -> int:
        broken_letters_set = set(broken_letters)

        words_count = 0

        for word in text.split():
            if set(word).intersection(broken_letters_set):
                continue

            words_count += 1

        return words_count
